azurelinuxagent/ga/update.py

```diff
@@ -489,11 +489,11 @@ def _try_update_goal_state(self, protocol):
             self._goal_state = protocol.get_goal_state()
 
             if self._update_goal_state_error_count > 0:
-                self._update_goal_state_error_count = 0
```

Comment:
This (tacitly) assumes the rest of the `if` block will not throw, but as per #2575, that isn't a valid assumption. Moving this line to the end of the block means that if we hit any more errors of the same type, this code will still work.